fxdc/config.py

```
@@ -37,17 +54,88 @@ def __init__(
+            if key not in newkwargs:
+                newkwargs[key] = value
+
+        # CHECKS
+        for key, value in newkwargs.items():
+            # Check Type Checking
+            if key in self.meta_data.get("typechecking", {}):
+                expected_type = self.meta_data["typechecking"][key]
+                if not isinstance(value, expected_type):
+                    raise TypeCheckFailure(
+                        f"Expected type {expected_type} for {key}, got {type(value)}"
+                    )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)